File: FlutterProjectKAMSOFT_client/flutterprojectkamsoft_client/lib/models/patient.dart
class Patient {
  final int cipherType;
  final PatientName name;
  final String pessel;
  final String dateOfBirth;
  final String diseaseDescription;
  final List<PatientAppointment> appointments;

  Patient({
    required this.cipherType,
    required this.name,
    required this.pessel,
    required this.dateOfBirth,
    required this.diseaseDescription,
    required this.appointments,
  });

  factory Patient.fromJson(Map<String, dynamic> json) {
    final appointmentsJson = json['appointments'] as List<dynamic>? ?? [];

    return Patient(
      cipherType: json['cipherType'] as int? ?? 0,
      name: PatientName.fromJson(json['name'] as Map<String, dynamic>),
      pessel: json['pessel']?.toString() ?? '',
      dateOfBirth: json['dateOfBirth']?.toString() ?? '',
      diseaseDescription: json['diseaseDescription']?.toString() ?? '',
      appointments: appointmentsJson
          .map((item) => PatientAppointment.fromJson(item as Map<String, dynamic>))
          .toList(),
    );
  }
}

class PatientName {
  final String firstName;
  final String lastName;

  PatientName({
    required this.firstName,
    required this.lastName,
  });

  factory PatientName.fromJson(Map<String, dynamic> json) {
    return PatientName(
      firstName: json['firstName'] as String,
      lastName: json['lastName'] as String,
    );
  }

  @override
  String toString() => '$firstName $lastName';
}

class PatientAppointment {
  final String appointmentDate;
  final String description;
  final String title;
  final String type;

  PatientAppointment({
    required this.appointmentDate,
    required this.description,
    required this.title,
    required this.type,
  });

  factory PatientAppointment.fromJson(Map<String, dynamic> json) {
    return PatientAppointment(
      appointmentDate: json['appointmentDate']?.toString() ?? '',
      description: json['description']?.toString() ?? '',
      title: json['title']?.toString() ?? '',
      type: json['type']?.toString() ?? '',
    );
  }
}

File: FlutterProjectKAMSOFT/Models/Patient.cs
using FlutterProjectKAMSOFT.Patterns.Strategy;

namespace FlutterProjectKAMSOFT.Models
{
    public class Patient
    {
        public PatientName Name { get; set; }
        public long PESSEL { get; set; }
        public DateOnly DateOfBirth { get; set; }
        public required List<Appointment> Appointments { get; set; }
        public IDiseaseStrategy Disease { get;  set; }
        public void WritePatientData()
        {
            Console.WriteLine($"First name: {Name.FirstName}");
            Console.WriteLine($"Last name: {Name.LastName}");
            Console.WriteLine($"PESSEL: {PESSEL}");
            Console.WriteLine($"Date of birth: {DateOfBirth.ToString()}");

            Console.WriteLine("______________________________________________________");
            for (int i = 0; i < Appointments.Count; i++)
            {
                Console.WriteLine($"Appointment {i + 1}:");
                Console.WriteLine($"Title: {Appointments[i].Title}");
                Console.WriteLine($"Description: {Appointments[i].Description}");
                Console.WriteLine($"Type: {Appointments[i].Type}");
                Console.WriteLine($"Date: {Appointments[i].AppointmentDate.ToString()}");
            }
        }
        public void SetDisease(IDiseaseStrategy disease)
        {
            Disease = disease;
        }
        public string GetPatientDisease()
        {
            if (Disease == null)
            {
                return "No disease assigned to the patient.";
            }
            return Disease.GetDiseaseDescription();
        }
    }
}
